mcdc/type_.py

```diff
@@ -1194,9 +1298,9 @@ def make_type_global(input_deck):
             ("gyration_radius", float64, (N_cycle,)),
             ("idx_cycle", int64),
             ("cycle_active", bool_),
-            ("eigenvalue_tally_nuSigmaF", float64),
-            ("eigenvalue_tally_n", float64),
-            ("eigenvalue_tally_C", float64),
+            ("eigenvalue_tally_nuSigmaF", float64, (1,)),
+            ("eigenvalue_tally_n", float64, (1,)),
+            ("eigenvalue_tally_C", float64, (1,)),
```

Comment:
Why is the third argument needed here but not on the other items that are also scalar?

Reply:
In order for fields to be safely added into across many threads without a mutex, we need to apply atomics.

Numba atomics work in terms of arrays, but not scalars, so they had to be changed to single-element arrays. This is rather silly, but it is probably the only option unless Numba itself changes.

Reply:
Got it. Let's put a comment about that. I think it is important to put a comment on why we particularly do certain things.